packages/core/src/analytics/integrations/Omnitracking/Omnitracking.js

```diff
@@ -215,26 +218,56 @@ class Omnitracking extends Integration {
         );
       }
       case analyticsTrackTypes.TRACK: {
-        precalculatedParameters = this.getPrecalculatedParametersForEvent(data);
+        const eventMapperFn = trackEventsMapper[data.event];
 
-        if (!this.validateTrackingRequisites()) {
-          logger.warn(
-            `[Omnitracking] - Event ${data.event} tried to track without an unique view id.
-                Consider tracking an event only after a page event.
-                This behaviour is not recommended and will not be supported in upcoming versions`,
+        if (eventMapperFn && !isFunction(eventMapperFn)) {
+          throw new Error(
+            '[Omnitracking] - The event mapper did not return a valid function. Please make sure the event is correctly mapped.',
           );
         }
 
-        return await this.postTracking(
-          formatTrackEvent(data, precalculatedParameters),
-          data,
-        );
+        if (eventMapperFn) {
+          const mappedEvents = eventMapperFn(data);
+
+          if (isArray(mappedEvents)) {
+            await Promise.all(
+              mappedEvents.map(async mappedEventData => {
+                await this.processTrackEvents(data, mappedEventData);
```

Comment:
I think this await might be unnecessary, because it will prevent the requests to be executed in parallel. Is there any problem in removing the `await` from this call and only returning the result of `this.processTrackEvents(data, mappedEventData)`? Without the `await` those events would arrive to the server out-of-order, so if it is necessary that they arrive in order, keep the `await` instruction.

Reply:
Fixed on dev-main-22-02-22. thanks 🙂